src/Host/Protocol/Impl/CustomHttpError.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Host.Protocol {
+    public enum CustomHttpError {
```

Comment:
By strings I don't mean localizable user-readable text. I mean sending a header like this:
```
RTVS-Broker-API-Error: InterpreterNotFound
```
and parsing it on the other side as such (`Enum.Parse`), instead of using numbers. Since the protocol is already text-based, there doesn't seem to be much value in using numeric values here.

As far as inventing our own range, there are still problems with custom codes. For one thing, future HTTP spec updates may add new codes in that range (it happened before), and ASP.NET Core might start using them for something else entirely; or `HttpClient` might add special handling for them. Basically, we have no guarantee that the rest of the stack won't stomp on any custom values we use here. Sticking to what the protocol does and doesn't guarantee ensures our future ability to use standard HTTP and REST frameworks.

Also, while the API is not really public yet, it is quite likely to be in the future. People might want to write e.g. watchdog monitoring scripts and such - think about Azure management APIs. So, while matters such as API stability are not an issue, we should try to keep it a clean RESTful API following the spec and best practices.